content/millennium/dstu2/scheduling/slot.md

```diff
@@ -52,25 +65,28 @@ Search for Schedules that meet supplied query parameters:
  Name             | Required?                                                        | Type           | Description
 ------------------|------------------------------------------------------------------|----------------|------------------------------------------------------------------------------------
 `_id`             | This, or `slot-type`                                             | [`token`]      | The logical resource id associated with the resource.
-`slot-type`       | Yes when using `schedule.actor` and\or `-location`, or `id`      | [`token`]      | The type of appointments that can be booked into this slot. Example: `12345`
-`schedule.actor`  | This and `slot-type`, or `id`                                    | [`reference`]  | A single or comma separated list of Practitioner references. Example: `Practitioner/12345`
-`-location`       | This and `slot-type`, or `id`                                    | [`reference`]  | A single or comma separated list of Location references. Example: `12345`
-`date`            | Yes when using `slot-type`                                       | [`date`]       | The Slot date-time. Example: `2016`
-[`_count`]        | No                                                               | [`number`]     | The maximum number of results to be returned per page. Defaults to `50`.
+`slot-type`       | Yes when using either `schedule.actor` or `-location`, or `_id`  | [`token`]      | The type of appointments that can be booked into this slot. Example: `http://snomed.info/sct|394581000`
```

Comment:
This _almost_ sounds like "slot-type is required with "_id", if you didn't already know the pattern.

This API has a lot of interdependent query parameters, and expressing the alternatives is a bit cumbersome as-is.  I'm thinking it may be helpful to separate out the "_id" parameter use-case so that the language for query parameters is simpler (i.e. removing the "or _id" from each line).  Then, if you're querying by parameters, you can say definitely that slot-type is required, date is required, and that either actor or location are required.

Reply:
Addressed with https://github.com/cerner/fhir.cerner.com/pull/170/commits/14a8b9797d173634ac02067d8219abea5946619f